tough-kms/src/lib.rs

```diff
@@ -0,0 +1,203 @@
+pub enum KmsSigningAlgorithm {
+    /// Signing Algorithm `RSASSA_PSS_SHA_256`
+    RsassaPssSha256,
+}
+
+impl Default for KmsSigningAlgorithm {
+    fn default() -> Self {
+        KmsSigningAlgorithm::RsassaPssSha256
+    }
+}
+
+impl KmsSigningAlgorithm {
+    fn value(self) -> String {
+        // Currently we are supporting only single algorithm, but code stub is added to support
+        // multiple algorithms in future.
+        match self {
+            #![allow(clippy::wildcard_in_or_patterns)]
+            #[allow(unreachable_patterns)]
+            KmsSigningAlgorithm::RsassaPssSha256 | _ => String::from("RSASSA_PSS_SHA_256"),
+        }
+    }
+}
+
+/// Implements the `KeySource` trait for keys that live in AWS KMS
+#[derive(Default)]
+pub struct KmsKeySource {
+    /// Identifies AWS account named profile, if not provided default AWS profile is used.
+    pub profile: Option<String>,
+    /// Identifies an asymmetric CMK in AWS KMS.
+    pub key_id: String,
+    /// KmsClient Object to query AWS KMS
+    pub client: Option<KmsClient>,
+    /// Signing Algorithm to be used for the message digest, only `KmsSigningAlgorithm::RsassaPssSha256` is supported at present.
+    pub signing_algorithm: KmsSigningAlgorithm,
+}
+
+impl fmt::Debug for KmsKeySource {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("KmsKeySource")
+            .field("key_id", &self.key_id)
+            .field("profile", &self.profile)
+            .finish()
+    }
+}
+
+/// Implement the `KeySource` trait.
+impl KeySource for KmsKeySource {
+    fn as_sign(
+        &self,
+    ) -> std::result::Result<Box<dyn Sign>, Box<dyn std::error::Error + Send + Sync + 'static>>
+    {
+        let kms_client = match self.client.to_owned() {
+            Some(value) => value,
+            None => client::build_client_kms(self.profile.as_deref())?,
+        };
+        // Get the public key from AWS KMS
+        let fut = kms_client.get_public_key(rusoto_kms::GetPublicKeyRequest {
+            key_id: self.key_id.clone(),
+            ..rusoto_kms::GetPublicKeyRequest::default()
+        });
+        let response = tokio::runtime::Runtime::new()
+            .context(error::RuntimeCreation)?
+            .block_on(fut)
+            .context(error::KmsGetPublicKey {
+                profile: self.profile.clone(),
+                key_id: self.key_id.clone(),
+            })?;
+        let key = pem::encode_config(
+            &pem::Pem {
+                tag: String::from("PUBLIC KEY"),
+                contents: response.public_key.context(error::PublicKeyNone)?.to_vec(),
+            },
+            pem::EncodeConfig {
+                line_ending: pem::LineEnding::LF,
+            },
+        );
+        if !response
+            .signing_algorithms
+            .context(error::MissingSignAlgorithm)?
+            .contains(&self.signing_algorithm.value())
+        {
+            error::ValidSignAlgorithm.fail()?;
+        }
```

Comment:
Prefer snafu's [`ensure!`](https://docs.rs/snafu/0.6.8/snafu/macro.ensure.html) macro.

```suggestion
        ensure!(
            !response
                .signing_algorithms
                .context(error::MissingSignAlgorithm)?
                .contains(&self.signing_algorithm.value()),
            error::ValidSignAlgorithm,
        );
```

Reply:
Done.